### PR TITLE
Added JSON server

### PR DESCRIPTION
**Pull request by WahabDev12**
### CHORE
Setup JSON server:  Issue #51 

### FEAT
* Created a dummy JSON server using the `json-server` npm package. Dummy data can be found in db.json in the root folder.
### SETTING UP
* First of all, install the `json-server` package using `npm install json-server`
* To start the json server locally, run the script `npm run json:server` in the root directory
* Navigate to `localhost:5000/<route>`  to display the results. eg: `localhost:5000/challenges`

### TL;DR
* I might have incorrectly represented the database schema.

### CODE REVIEW

**Review by faddalibrahim — COMMENTED** [review]
the hackers and challenge are sub-documents in seasons. they are not top level

I am guessing JSON-server requires the objects to be top level? no nested objects?
